packages/compass-sidebar/src/components/connections-filter-popover.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useCallback, type PropsWithChildren } from 'react';
+
+import {
+  css,
+  Icon,
+  IconButton,
+  InteractivePopover,
+  Label,
+  Overline,
+  palette,
+  spacing,
+  Toggle,
+  Tooltip,
+  useId,
+} from '@mongodb-js/compass-components';
+import type { ConnectionsFilter } from './use-filtered-connections';
+
+const containerStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'flex-start',
+  padding: spacing[300],
+  minWidth: 270,
+});
+
+const closeButtonStyles = css({
+  // An alternative to this is to pass hideCloseButton to InteractivePopover,
+  // but that throws an error when the popover is opened
+  display: 'none',
+});
+
+const activatedIndicatorStyles = css({
+  position: 'absolute',
+  top: spacing[50],
+  right: spacing[50],
+});
+
+const groupStyles = css({
+  display: 'flex',
+  flexDirection: 'row',
+  gap: spacing[200],
+  marginTop: spacing[200],
+});
+
+type ConnectionsFilterPopoverProps = PropsWithChildren<{
+  open: boolean;
+  setOpen: (open: boolean) => void;
+  filter: ConnectionsFilter;
+  onFilterChange(
+    updater: (filter: ConnectionsFilter) => ConnectionsFilter
+  ): void;
+}>;
+
+export default function ConnectionsFilterPopover({
+  open,
+  setOpen,
+  filter,
+  onFilterChange,
+}: ConnectionsFilterPopoverProps) {
+  const onExcludeInactiveChange = useCallback(
+    (excludeInactive: boolean) => {
+      onFilterChange((filter) => ({
+        ...filter,
+        excludeInactive,
+      }));
+    },
+    [onFilterChange]
+  );
+
+  const excludeInactiveToggleId = useId('exclude-inactive-toggle');
```

Comment:
`useId` with default argument just returns the argument you are passing, there is no reason to use it here if you're ids are constant

Reply:
Looking at use elsewhere in the codebase, I was under the impression that this would add some suffix to avoid conflicts if the component is mounted multiple places in the tree? But apparently I was wrong in that assumption 😬 Thanks.

Reply:
Yeah, good point, seems like a few misuses slipped through, opene https://github.com/mongodb-js/compass/pull/6491 to fix those